src/DistributedLock.Etcd/EtcdLeaseDistributedLockHandle.cs

```diff
@@ -0,0 +1,44 @@
+using dotnet_etcd.interfaces;
+using Etcdserverpb;
+using Medallion.Threading.Internal;
+
+namespace Medallion.Threading.Etcd;
+
+public class EtcdLeaseDistributedLockHandle : IDistributedSynchronizationHandle
+{
+    public string Key;
+
+    internal EtcdLeaseDistributedLockHandle(string key, EtcdClientWrapper client, long leaseKey)
+    {
+        this.Key = key ?? throw new ArgumentNullException(nameof(key));
+        this.Client = client ?? throw new ArgumentNullException(nameof(client));
+        this.LeaseKey = leaseKey;
+        // Because this is a lease, managed finalization mostly won't be strictly necessary here. Where it comes in handy is:
+        // (1) Ensuring blob deletion if we own the blob
+        // (2) Helping release infinite-duration leases (rare case)
+        // (3) In testing, avoiding having to wait 15+ seconds for lease expiration
```

Comment:
This comment seems copied from AzureBlobLeaseDistributedLockHandle. I think it mostly applies here EXCEPT that we're currently allowing infinite TTL leases in which case managed finalization would be needed. My guess is that the answer is that we shouldn't support infinite ttl leases